zebra-chain/src/parameters/network.rs

```diff
@@ -85,11 +170,26 @@ impl fmt::Display for Network {
 }
 
 impl Network {
+    /// Creates a new `Testnet` with no parameters.
+    pub fn new_testnet() -> Self {
```

Comment:
It looks like we're using this method for both:
- the canonical public testnet used in tests
- the custom testnet parameters used by `zebrad` when configured with Testnet

They can be the same, but we also want to make it easy for them to be different. And make the code easy to maintain.

So here's what I suggest to make the code clearer and easier to maintain:
- use a PUBLIC_TESTNET constant in tests, and make it test-only/proptest-impl using #cfg
- use a new_configured_testnet() method in production code, it should only be called once when the network config is loaded, so it can be pub(crate)
- make all the other methods that can create a testnet into test-only/proptest-impl using #cfg 

After that change, anything that doesn't compile is an incorrect use of the public testnet in production code, or an incorrect use of the configured testnet in test code.

We'll also need tests for custom parameter changes, so we can't make those methods private, they need to be called from tests.

```suggestion
    pub fn new_public_testnet() -> Self {
```